test/unit/test-amp-img.js

```diff
@@ -665,4 +666,121 @@ describe('amp-img', () => {
       );
     });
   });
+
+  // Firefox misbehaves on Windows for this test because getBoundingClientRect
+  // returns 0x0 for width and height. Strangely Firefox on MacOS will return
+  // reasonable values for getBoundingClientRect if we add an explicit wait
+  // for laid out attributes via waitForElementLayout. If we change the test to
+  // test for client or offset values, Safari yields 0px measurements.
+  // For details, see: https://github.com/ampproject/amphtml/pull/24574
+  describe
+    .configure()
+    .skipFirefox()
+    .run('layout intrinsic', () => {
+      let browser;
+      beforeEach(() => {
+        fixture.iframe.height = 800;
+        fixture.iframe.width = 800;
+        browser = new BrowserController(fixture.win);
+      });
+      it('should not exceed given width and height even if image\
+      natural size is larger', () => {
+        let ampImg;
+        return getImg({
+          src: '/examples/img/sample.jpg', // 641 x 481
+          width: 100,
+          height: 100,
+          layout: 'intrinsic',
+        })
+          .then(image => {
+            ampImg = image;
+            return browser.waitForElementLayout('amp-img');
+          })
+          .then(() => {
+            const dimensions = ampImg.getBoundingClientRect();
```

Comment:
A bit cleaner to have these as:

```javascript
exepct(ampImg.getBoundingClientRect()).to.include({
  width: 100,
  height: 100,
});
```

For example like here: https://github.com/ampproject/amphtml/blob/a423cabb75b3431de39c52249717734263953ded/extensions/amp-truncate-text/0.1/test/test-amp-truncate-text.js#L422-L425

The benefit is that if one or both of the properties fail, you get a single message and you can see what are the other rect properties in the error message. In general, it is better to have a single expectation to cover this, for example if the test fails on height, you make some changes to fix that, then the test fails on width.